return {
    PlayerComponentConfig = require(script.PlayerComponentConfig),
    Player = require(script.Player),
    PlayerBlackboard = require(script.PlayerBlackboard),
    PlayerStateMachine = require(script.PlayerStateMachine),
    HealthComponent = require(script.Components.HealthComponent),
    StaminaComponent = require(script.Components.StaminaComponent),
    MovementComponent = require(script.Components.MovementComponent),
    InventoryComponent = require(script.Components.InventoryComponent),
}
