return {
    ControlStateService = require(script.ControlStateService),
    FormalLocomotion = require(script.FormalLocomotion),
    HexMazeWorldRenderer = require(script.HexMazeWorldRenderer),
    InventoryService = require(script.InventoryService),
    MazeDebugWorldComposer = require(script.MazeDebugWorldComposer),
    MazeFormalWorldComposer = require(script.MazeFormalWorldComposer),
    MazeLightingProfile = require(script.MazeLightingProfile),
    MazeModuleAssetContract = require(script.MazeModuleAssetContract),
    MazeWorldShellBuilder = require(script.MazeWorldShellBuilder),
    MonsterRuntime = require(script.MonsterRuntime),
    MonsterService = require(script.MonsterService),
    PlayerStateService = require(script.PlayerStateService),
    TeleportInventoryHydrator = require(script.TeleportInventoryHydrator),
}
