return {
    ControlStateService = require(script.ControlStateService),
    HexMazeWorldRenderer = require(script.HexMazeWorldRenderer),
    InventoryService = require(script.InventoryService),
    MazeDebugWorldComposer = require(script.MazeDebugWorldComposer),
    MazeFormalWorldComposer = require(script.MazeFormalWorldComposer),
    MazeModuleAssetContract = require(script.MazeModuleAssetContract),
    MazeWorldShellBuilder = require(script.MazeWorldShellBuilder),
    MonsterService = require(script.MonsterService),
    PlayerStateService = require(script.PlayerStateService),
}
